### DISCUSSION TITLE
Should we add a Feature matrix to CFI

### DISCUSSION THREAD

**Discussion by thinkl33t** [finos/compliant-financial-infrastructure#208] [Ideas] [answered]
I've been working on-and-off on a feature matrix for CFI, which is designed to offer an at-a-glance view of the level of maturity of the various services we provide.  This is now a little outdated, but shows the kind of thing i'm thinking of:

In the future i see this as being automateable, with the icons themselves linking to documentation, implementation, and testing evidence, but initially it can be set by a maintainer as part of a merge request approval.

Do people think this would be a valuable thing to have as part of CFI?

---

# CFI Feature Matrix

CFI provides multiple services, across multiple clouds.  These services range from not-yet-implemented, to fully featured, and automatically tested for compliance with the CDMC framework.

This feature matrix is intended to show all current and planned services along with the current status of each, to help contributors looking for a task know where best to spend their effort.

## Feature Matrix

|                    |         AWS         |         GCP         |        Azure        |    OpenShift     |
| ------------------:|:-------------------:|:-------------------:|:-------------------:|:----------------:|
| Kubernetes Cluster | 🥇 | 🥉 |  📄   | 📄 |
|  Postgres Database |                     |                     | 🥉 |                  |
|           DynamoDB |  📄  |         N/A         |         N/A         |       N/A        |
|           RedShift |  📄   |         N/A         |         N/A         |       N/A        |
|                SQS |  📄  |         N/A         |         N/A         |       N/A        |

### 💎 diamond
Services rated diamond will meet all the specifications of lower tiers of system, and will produce CI output proving they pass compliance guidelines. 

### 🥇 gold
Tests are automatically run in CI as part of the testing phase, though some may fail

### 🥈 silver
The service is automatically smoke tested before being committed to `main`, through being spun up and destroyed via CI

### 🥉 bronze
Infrastructure-as-code has been produced that is able to create and destroy the service, where said service meets the SAA specification

### 📄 paper
A Service Approval Accelerator document has been produced for this service, and accepted into `main`.  The specification and tests in this SAA should meet the guidelines of our compliance framework.

**Comment by eddie-knight** [comment, marked as answer]
Yes. In my mind at least... this is already a thing. Next step is to get a workflow/process formalized and create the appropriate docs in CFI.

**Comment by thinkl33t** [comment]
We have done :)
